src/domain/products/batch-job/import.tsx

```diff
@@ -0,0 +1,180 @@
+  const isPreprocessed = !!batchJob?.result
+  const progress = isPreprocessed
+    ? batchJob!.result.advancement_count / batchJob!.result.count
+    : undefined
+
+  const status = isPreprocessed
+    ? undefined
+    : isUploaded
+    ? "Preprocessing..."
+    : "Uploading..."
+
+  /**
+   * Confirm job on submit.
+   */
+  const onSubmit = async () => {
+    await confirmBatchJob()
+    notification(
+      "Success",
+      "Import confirmed for processing. Progress info is available in the activity drawer.",
+      "success"
+    )
+    props.handleClose()
+  }
+
+  /**
+   * Upload file and use returned file key to create a batch job.
+   */
+  const processUpload = async (file: File) => {
+    try {
+      const res = await uploadFile(file as any)
+      const _fileKey = res.uploads[0].key
+      setFileKey(_fileKey)
+
+      const batchJob = await createBatchJob({
+        dry_run: true,
+        context: { fileKey: _fileKey },
+        type: "product-import",
+      })
+
+      setBatchJobId(batchJob.batch_job.id)
+    } catch (e) {
+      notification("Error", "Import failed.", "error")
+      if (fileKey) {
+        await deleteFile({ file_key: fileKey })
+      }
+    }
+  }
+
+  /**
+   * Returns create/update counts from stat descriptor.
+   */
+  const getSummary = () => {
+    if (!batchJob) {
+      return undefined
+    }
+
+    const res = batchJob.result?.stat_descriptors[0].message.match(/\d+/g)
+
+    if (!res) {
+      return undefined
+    }
+
+    return {
+      toCreate: Number(res[0]),
+      toUpdate: Number(res[1]),
+    }
+  }
+
+  /**
+   * When file upload is removed, delete file from the bucket and cancel batch job.
+   */
+  const onFileRemove = async () => {
+    try {
+      // TODO: fix this function has uncaught error
+      deleteFile(fileKey!)
+      cancelBathJob()
+    } catch (e) {
+      console.log(e)
+    }
+  }
+
+  /**
+   * Cleanup file if batch job isn't confirmed.
+   */
+  const onClose = () => {
+    if (
+      !["confirmed", "completed", "canceled", "failed"].includes(
+        batchJob?.status
+      )
+    ) {
+      if (fileKey) {
+        deleteFile({ file_key: fileKey })
+      }
+      if (batchJobId) {
+        cancelBathJob()
+      }
+    }
+  }
+
+  return (
+    <UploadModal
+      status={status}
+      progress={progress}
+      canImport={isPreprocessed}
+      onSubmit={onSubmit}
+      onClose={onClose}
+      summary={getSummary()}
+      onFileRemove={onFileRemove}
+      processUpload={processUpload}
+      fileTitle={"products list"}
+      description2Title="Unsure about how to arrange your list?"
+      description2Text="Download the template below to ensure you are following the correct format."
+      description1Text="Through imports you can add or update products. To update existing products you must use the existing handle, to update existing variants you must use the existing SKU. You will be asked for confirmation before we import products."
```

Comment:
comment: this text should be changed since we are using existing ids instead of SKU/handle